### PR TITLE
Refine API

### PR DESCRIPTION
**Pull request by burke**
Implement https://github.com/Shopify/cli-kit/pull/13

### CODE REVIEW

File: lib/todo/executor.rb
Comment (lugray) [inline lib/todo/executor.rb:4]: This thing we did with Config to work around new and old ways feels like an anti-pattern that I'd rather not perpetuate.

Comment (jules2689) [reply lib/todo/executor.rb:4]: I feel like this could be a default or something, I'm having a hard time following the api split here

Comment (burke) [reply lib/todo/executor.rb:4]: How should we accomplish this?

* We have a module of functionality, the `Executor`.
* It needs three bits of information provided:
    * Name of tool
    * Command Registry
    * Log file

Options I can think of:

* Subclass in app, require redefinition of these three methods.
* Statically assign these values in the `CLI::Kit` singleton instance
* Require application to pass configuration values to each API call on the Executor, rather than on on initialization

---

We talked about this a bunch in person and have come to a rough consensus that this is a probably-reasonable goal, but the implementation still feels a little off because, while `Executor` isn't truly a class or module, it is masquerading as one (although it is fair to treat it as a module because of the way it is). We discussed assigning it in screaming snake case, but that makes one wonder whether it's a module or a primitive, which is another tradeoff. Assigning it as a class method on `Todo` is also not a lie, but `Todo.executor.call` doesn't convey the static-ness of `Todo::Executor`. I'll try to find a better way.

File: exe/todo
Comment (jules2689) [inline exe/todo:15]: I think we can uncomment this now?

Comment (burke) [reply exe/todo:15]: I should actually make it work. The API here is not very clean. That would work if we had an `ErrorReporter`

**Review by jules2689 — APPROVED** [review]
I think we should merge this and iterate on it further if needed. There's a lot here and I would rather see how things feel as we go than to try and nail every nail right now

**Comment by burke** [comment]
Good. Once this is merged I'm going to bring cli-kit-example back into cli-kit as a generator.

**Pull request merged**
